### PR TITLE
Oxidation merge types

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

This PR was discussed with @touilleMan, the idea is to simplify `libparsec_types` and `libparsec_client_types`, because most of time we use both crates together (excepted in `libparsec_protocol` and `libparsec_client_connection`)

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: oxidation/libparsec/crates/types/tests/mod.rs [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 
+// Functions using rstest parametrize ignores `#[warn(clippy::too_many_arguments)]`
+// decorator, so we must do global ignore instead :(
+#![allow(clippy::too_many_arguments)]
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/types/tests/mod.rs:5]: Maybe it's the indication that we need to define/use a struct :shrug:

Comment (TimeEngineer) [reply oxidation/libparsec/crates/types/tests/mod.rs:5]: Yes but out of scope

Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/types/tests/mod.rs:5]: Open an issue

Comment (TimeEngineer) [reply oxidation/libparsec/crates/types/tests/mod.rs:5]: https://github.com/Scille/parsec-cloud/issues/4371

**Pull request merged**